runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/stream/HttpClientFactory.java

```diff
@@ -2421,6 +2421,10 @@ else if (beginEx == null &&
                      pool.versions.contains(HTTP_2))
             {
                 this.encoder = HttpEncoder.H2C;
+                for (HttpExchange exchange: pool.exchanges.values())
+                {
+                    exchange.remoteBudget += encodeMax;
+                }
```

Comment:
I'd like to understand what is driving this change and also why it should differ from the case above it regarding `flushRequestWindow`.

Reply:
Not sure I understand 
```
also why it should differ from the case above it regarding flushRequestWindow.
```
You mean why there is no flushRequestWindow in this case. 

I believe in case of http2 

```
for (HttpExchange exchange: pool.exchanges.values())
                {
                    exchange.remoteBudget += encodeMax;
                    exchange.flushRequestWindow(traceId, 0);
                }
```

Calling  `exchange.flushRequestWindow(traceId, 0);` is mostly driven by our existing server specs so that we can start receiving data as soon as possible without modifying the scripts.

In the case of h2c,  the problem was by spec we are supposed to send payload along with headers before receiving any data from the server if we don't set the remote max then what happens is that we are going to receive reply data before sending any data.


```
read "POST / HTTP/1.1\r\n"
     "Host: localhost:8080\r\n"
     "Content-Type: text/plain;charset=UTF-8\r\n"
     "Content-Length: 12\r\n"
     "Upgrade: h2c\r\n"
     "Connection: Upgrade, HTTP2-Settings\r\n"
     "Http2-Settings: AAMAAABkAAQAAP__\r\n"
     "\r\n"

read "Hello, world"

write "HTTP/1.1 101 Switching Protocols\r\n"
      "Connection: Upgrade\r\n"
      "Upgrade: h2c\r\n"
      "\r\n"
`````